pandas/tseries/offsets.py

```diff
@@ -1377,7 +1386,7 @@ class Week(EndMixin, DateOffset):
     _prefix = 'W'
 
     def __init__(self, n=1, normalize=False, weekday=None):
-        self.n = n
+        self.n = int(n)
```

Comment:
add a ``self.n = self._validate(n)`` method to do this, you want to give a much more informative message, so add tests for this as well (e.g. try passing 'a' or something in), further we for sure want to maybe warn if a non-integer float is passed in (this may break things so have maybe do as a followup). offset creation is user facing so it needs to be infromative.